airbyte-commons-worker/src/main/java/io/airbyte/workers/general/DefaultReplicationWorker.java

```diff
@@ -340,16 +342,30 @@ private static Runnable readFromSrcAndWriteToDstRunnable(final AirbyteSource sou
         populatedStreamToSelectedFields(catalog, streamToSelectedFields);
       }
       try {
+        final Duration MAX_FETCH_SECONDS = Duration.ofSeconds(30);
+        long lastMessageRecieved = System.currentTimeMillis();
         while (!cancelled.get() && !source.isFinished()) {
           final Optional<AirbyteMessage> messageOptional;
+          final Callable<Optional<AirbyteMessage>> task = () -> source.attemptRead();
+
+          final Duration durationSinceLast = getDurationSinceLast(lastMessageRecieved, System.currentTimeMillis());
+
+          final Future<Optional<AirbyteMessage>> future = executors.submit(task);
```

Comment:
how much performance overhead does this add?

Reply:
I'm going to prepare load test on a dev env